bindings/cpp/src/lib.rs

```diff
@@ -65,10 +74,16 @@ mod ffi {
         fn remove(self: &Operator, path: &str) -> Result<()>;
         fn stat(self: &Operator, path: &str) -> Result<Metadata>;
         fn list(self: &Operator, path: &str) -> Result<Vec<Entry>>;
+        fn reader(self: &Operator, path: &str) -> Result<Box<Reader>>;
+
+        #[cxx_name = "read"]
+        fn reader_read(self: &mut Reader, buf: &mut [u8]) -> Result<usize>;
```

Comment:
Maybe we should split them in two different namespaces?

Reply:
We move `Reader` into another file. This can avoid name conflicting.